torch/csrc/jit/script/compiler.cpp

```diff
@@ -1180,20 +1206,25 @@ struct to_ir {
     std::set<std::string> mutated_variables;
 
     for (auto& v : save_true->definedVariables()) {
-      if (save_false->findInAnyFrame(v)) {
+      if (save_false->findInAnyFrame(v) || false_exits) {
         mutated_variables.insert(v);
       }
     }
     for (auto& v : save_false->definedVariables()) {
-      if (save_true->findInAnyFrame(v)) {
+      if (save_true->findInAnyFrame(v) || true_exits) {
         mutated_variables.insert(v);
       }
     }
 
+    WithInsertPoint guard(graph->block()->nodes().front());
+    auto b_val = graph->insertNode(graph->create(prim::Bottom, {}, 1))
```

Comment:
prim::Constant?

Reply:
Why not just create one of these once at the beginning of the pass?

Reply:
prim::Constant : All other constant values are generated by the prim::Constant node (e.g. there is no prim::None node to generate a None type). Same should happen for the Bottom type.